src/core/lombok/eclipse/HandlerLibrary.java

```diff
@@ -219,9 +219,15 @@ public long handleAnnotation(CompilationUnitDeclaration ast, EclipseNode annotat
 		TypeResolver resolver = new TypeResolver(annotationNode.getImportList());
 		TypeReference rawType = annotation.type;
 		if (rawType == null) return Long.MAX_VALUE;
-		
+
 		String fqn = resolver.typeRefToFullyQualifiedName(annotationNode, typeLibrary, toQualifiedName(annotation.type.getTypeName()));
-		if (fqn == null) return Long.MAX_VALUE;
+		if (fqn == null) {
+			if (isNonNullAnnotation(annotationNode, annotation)) {
+				fqn = lombok.NonNull.class.getName();
```

Comment:
This fixes the inconsistent behaviour mentioned in https://github.com/projectlombok/lombok/issues/3305#issuecomment-1765993862

We run in this code path, if an unknown annotantion is found on an method or constructor paramerter. In this case we check, if this could be an alternative nonNull annotation. If so, we treat it like the lombok.NonNull annotation.